client/ayon_core/host/interfaces/workfiles.py

```diff
@@ -0,0 +1,1177 @@
+        all host integrations.
+
+    """
+    @abstractmethod
+    def save_workfile(self, dst_path: Optional[str] = None):
+        """Save the currently opened scene.
+
+        Args:
+            dst_path (str): Where the current scene should be saved. Or use
+                the current path if 'None' is passed.
+
+        """
+        pass
+
+    @abstractmethod
+    def open_workfile(self, filepath: str):
+        """Open passed filepath in the host.
+
+        Args:
+            filepath (str): Path to workfile.
+
+        """
+        pass
+
+    @abstractmethod
+    def get_current_workfile(self) -> Optional[str]:
+        """Retrieve a path to current opened file.
+
+        Returns:
+            Optional[str]: Path to the file which is currently opened. None if
+                nothing is opened.
+
+        """
+        return None
+
+    def workfile_has_unsaved_changes(self) -> Optional[bool]:
+        """Currently opened scene is saved.
+
+        Not all hosts can know if the current scene is saved because the API
+            of DCC does not support it.
+
+        Returns:
+            Optional[bool]: True if scene is saved and False if has unsaved
+                modifications. None if can't tell if workfiles has
+                modifications.
+
+        """
+        return None
+
+    def get_workfile_extensions(self) -> list[str]:
+        """Extensions that can be used to save the workfile to.
+
+        Notes:
+            Method may not be used if 'list_workfiles' and
+                'list_published_workfiles' are re-implemented with different
+                logic.
+
+        Returns:
+            list[str]: List of extensions that can be used for saving.
+
+        """
+        return []
+
+    def save_workfile_with_context(
+        self,
+        filepath: str,
+        folder_entity: dict[str, Any],
+        task_entity: dict[str, Any],
+        *,
+        version: Optional[int],
+        comment: Optional[str] = None,
+        description: Optional[str] = None,
+        rootless_path: Optional[str] = None,
+        workfile_entities: Optional[list[dict[str, Any]]] = None,
+        project_settings: Optional[dict[str, Any]] = None,
+        project_entity: Optional[dict[str, Any]] = None,
+        anatomy: Optional["Anatomy"] = None,
+    ):
+        """Save the current workfile with context.
+
+        Notes:
+            Should this method care about context change?
+
+        Args:
+            filepath (str): Where the current scene should be saved.
+            folder_entity (dict[str, Any]): Folder entity.
+            task_entity (dict[str, Any]): Task entity.
+            version (Optional[int]): Version of the workfile.
+            comment (Optional[str]): Comment for the workfile.
+            description (Optional[str]): Description for the workfile.
+            rootless_path (Optional[str]): Rootless path of the workfile.
+            workfile_entities (Optional[list[dict[str, Any]]]): Workfile
```

Comment:
What's this? What type of workfile entities would I pass to `save_workfile`... and why is it plural?

Reply:
I'm still a bit confused by this one actually - I see WHY you're passing it the list, because `ayon_api` allows you to query it that way. However, functionality wise I'd actually expect to query the workfile entity for a particular file (instead of ALL workfile entities for the task) just by the filepath or rootless filepath.

So that it'd just be `ayon_api.get_workfile_entity(project_name, rootless_path)` or alike?

It just looks confusing to pass this list even though it really only ever just 'guesses' and takes only ONE of the entries. It's never about taking multiples from that list. It's like passing an optimized lookup instead of saying, here take this workfile entity to set it on (which is what I'd have expected).

Reply:
True, but it does not support it now. We would have to wait for server PR, make server release, then change ayon api, make release, then use new ayon api in AYON launcher, make release, and then we can use it... 🤔

Reply:
At the same time - if we don't... and release the API like this instead - we'll have to maintain that backwards compatibility forever. Whereas if we clean up this one bit, the calls would be faster and the API we want to release much simplified and have less technical debt.

It _could_ be worth it if we want to produce a clean API for this. (and perhaps rely on graphql in the meantime?)

Reply:
> At the same time - if we don't... and release the API like this instead - we'll have to maintain that backwards compatibility forever. Whereas if we clean up this one bit, the calls would be faster and the API we want to release much simplified and have less technical debt.
> 
> It could be worth it if we want to produce a clean API for this. (and perhaps rely on graphql in the meantime?)

So you suggest to wait with this PR until it's available, which might be few weeks?